### PR TITLE
progress on #727

### PR DESCRIPTION
**Pull request by makc-dataloop**
@RotemDl could you please check if this works for you - code so far, and [the video in a comment](https://github.com/dataloop-ai/components/issues/727#issuecomment-1820713752)

### CODE REVIEW

File: src/components/compound/DlSearches/DlSmartSearch/utils/index.ts
Comment (fadiDL) [inline src/components/compound/DlSearches/DlSmartSearch/utils/index.ts:1]: @makc-dataloop why make a new set of functions that handle aliases setting and reverting? 
why not use the functions beforehand? 
This feels to me that it will over complicate the use of the functions 

if you want t o split the code here in the util then it's possible but id prefer that we had these new value alias handlers embedded into the original set and revert aliases functions.

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/components/compound/DlSearches/DlSmartSearch/utils/index.ts
Comment (makc-dataloop) [reply src/components/compound/DlSearches/DlSmartSearch/utils/index.ts:1]: well the ones we have - I assume you mean revertAliases and setAliases right above - are basically string search and replace. these new ones traverse the json instead and replace values of the specific fields only. so that the alias for field1 value would not replace the value of field2 in the same breath

File: src/hooks/use-suggestions.ts [hunk 1]
```diff
@@ -402,7 +402,7 @@ const getError = (
 
 const isValidByDataType = (
     str: string | string[],
-    dataType: string | string[],
+    dataType: string | any[],
```
Comment (fadiDL) [inline src/hooks/use-suggestions.ts:405]: why any[] ?

File: src/components/compound/DlSearches/DlSmartSearch/utils/index.ts
Comment (fadiDL) [inline src/components/compound/DlSearches/DlSmartSearch/utils/index.ts:1]: This needs tests to make sure everything is ok. either in the searchInput or as the util itself. 
please see where we tested the previous aliases functions

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/hooks/use-suggestions.ts (hunk 1 above)
Comment (makc-dataloop) [reply src/hooks/use-suggestions.ts:405]: well because of [this](https://github.com/dataloop-ai/components/pull/740/files#diff-1ead159cbb5550124194e698592bfd3b4f8963b5bfb41e5eef9e881481a9e2d9R143)

Comment (fadiDL) [reply src/hooks/use-suggestions.ts:405]: i still dont understand. we can easily determine the type needed without the need to use any
Can you change it to the type its supposed to accept?

Comment (makc-dataloop) [reply src/hooks/use-suggestions.ts:405]: @fadiDL because rotem wanted to put a dictionary into scheme which had string[] before - now it is not not an array of strings but an array of strings or objects. do you want me to type it as (string | object)[] instead of any[] here?

Comment (fadiDL) [reply src/hooks/use-suggestions.ts:405]: > @fadiDL because rotem wanted to put a dictionary into scheme which had string[] before - now it is not not an array of strings but an array of strings or objects. do you want me to type it as (string | object)[] instead of any[] here?

type it as 

(string | {[key: string]: any})[]

File: src/components/compound/DlSearches/DlSmartSearch/utils/index.ts
Comment (makc-dataloop) [reply src/components/compound/DlSearches/DlSmartSearch/utils/index.ts:1]: added tests to both useSuggestions and DlSmartSearchInput

File: src/hooks/use-suggestions.ts [hunk 2]
```diff
@@ -666,13 +679,22 @@ export const removeLeadingExpression = (str: string) => {
...
+) => {
+    const types: (string | Data)[] = Array.isArray(dataType)
+        ? dataType
+        : [dataType]
     const suggestion: string[] = []
 
     if (Array.isArray(dataType)) {
         suggestion.push(
-            ...dataType.filter((type) => !knownDataTypes.includes(type))
+            ...(dataType.filter(
```
Comment (fadiDL) [inline src/hooks/use-suggestions.ts:693]: @makc-dataloop i would prefer to rephrase this to a normal for-of loop and push to the array without the spread operator. 
Spreading of alot of elements can cause a stackoverflow error

**Review by fadiDL — APPROVED** [review]
